iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/ThreeWindingsTransformerImpl.java

```diff
@@ -386,24 +386,30 @@ public Terminal getTerminal(ThreeSides side) {
     @Override
     public Terminal getTerminal(String voltageLevelId) {
         Objects.requireNonNull(voltageLevelId);
-        boolean side1 = getLeg1().getTerminal().getVoltageLevel().getId().equals(voltageLevelId);
-        boolean side2 = getLeg2().getTerminal().getVoltageLevel().getId().equals(voltageLevelId);
-        boolean side3 = getLeg3().getTerminal().getVoltageLevel().getId().equals(voltageLevelId);
-        if (side1 && side2 && side3) {
+        boolean isLeg1ConnectedToVoltageLevel = isLegOnVoltageLevel(getLeg1(), voltageLevelId);
```

Comment:
Your variables' names are more precise than my suggested method's name.
It would be better to keep this precision by renaming the method `isLegConnectedToVoltageLevel`.